src/mqtt_client.rs

```diff
@@ -514,6 +522,33 @@ impl<
                     .handle_pubrec(rec.packet_id, packet)?;
             }
 
+            ReceivedPacket::PubRel(rel) => {
+                let session_state = &mut self.sm.context_mut().session_state;
+
+                let reason = if let Some(position) = session_state
+                    .pending_packets
+                    .iter()
+                    .position(|&id| id == rel.packet_id)
+                {
+                    session_state.pending_packets.swap_remove(position);
+                    info!("Remaining: {}", session_state.pending_packets.len());
+                    ReasonCode::Success
+                } else {
+                    ReasonCode::PacketIdNotFound
+                };
+
+                // Send a PubComp
+                let pubcomp = PubComp {
+                    packet_id: rel.packet_id,
+
+                    // Note: Because we do not support ExactlyOnce, it's not possible for
```

Comment:
I thought this implements ExactlyOnce.

Reply:
This must be a copy-paste fail. I'll take a look at this and figure out what the comment is referring to